ntbs-service/Migrations/20190829150251_AddClinicalTimelineFlags.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace ntbs_service.Migrations
+{
+    public partial class AddClinicalTimelineFlags : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "ClinicalTimelines",
+                columns: table => new
+                {
+                    NotificationId = table.Column<int>(nullable: false),
+                    SymptomStartDate = table.Column<DateTime>(nullable: true),
+                    PresentationDate = table.Column<DateTime>(nullable: true),
+                    DiagnosisDate = table.Column<DateTime>(nullable: true),
+                    TreatmentStartDate = table.Column<DateTime>(nullable: true),
+                    DeathDate = table.Column<DateTime>(nullable: true),
+                    DidNotStartTreatment = table.Column<bool>(nullable: false),
+                    IsPostMortem = table.Column<bool>(nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_ClinicalTimelines", x => x.NotificationId);
+                    table.ForeignKey(
+                        name: "FK_ClinicalTimelines_Notification_NotificationId",
+                        column: x => x.NotificationId,
+                        principalTable: "Notification",
+                        principalColumn: "NotificationId",
+                        onDelete: ReferentialAction.Cascade);
+                });

```

Comment:
Yes it's pretty neat, you can have "Owned" entities that don't have their own primary keys, and are basically just a collection of related properties for convenience purposes. They can either be in the same table (by default) or in separate tables (as I've done here).

I definitely think this will be applicable for other entities too - anywhere where we'd never want to search for an entity directly, without its parent (notification in this case). You can't do something like context.ClinicalTimeline... for owned entities, as the you can't create a db set for owned types (see https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities) - so if we ever want to fetch patient details without first fetching notifications, it might not be a good idea for that?

Reply:
Oh I just remember that we want to restructure our patient model, as we're making everything notification centric.. so I think patientdetails will be suited to be an owned type. Let's chat in person :)

Reply:
Great, as discussed let's move in that direction :)
I don't think we're interested in fetching e.g. all clinical timelines and not the rest of the notification models.